diff --git a/node_modules/expo-modules-core/build/registerWebModule.d.ts b/node_modules/expo-modules-core/build/registerWebModule.d.ts
index 3998d1b..3fe6c3a 100644
--- a/node_modules/expo-modules-core/build/registerWebModule.d.ts
+++ b/node_modules/expo-modules-core/build/registerWebModule.d.ts
@@ -4,5 +4,8 @@ import type { NativeModule } from './ts-declarations/NativeModule';
  * @param moduleImplementation – a class that extends `NativeModule`. The class is registered under `globalThis.expo.modules[className]`.
  * @returns A singleton instance of the class passed into arguments.
  */
-export declare function registerWebModule<EventsMap extends Record<never, never>, ModuleType extends typeof NativeModule<EventsMap>>(moduleImplementation: ModuleType): ModuleType;
+type ModuleIdentifier = string | {
+    name?: string;
+};
+export declare function registerWebModule<EventsMap extends Record<never, never>, ModuleType extends typeof NativeModule<EventsMap>>(moduleImplementation: ModuleType, moduleIdentifier?: ModuleIdentifier): ModuleType;
 //# sourceMappingURL=registerWebModule.d.ts.map
\ No newline at end of file
diff --git a/node_modules/expo-modules-core/src/registerWebModule.ts b/node_modules/expo-modules-core/src/registerWebModule.ts
index eaee38e..70495fc 100644
--- a/node_modules/expo-modules-core/src/registerWebModule.ts
+++ b/node_modules/expo-modules-core/src/registerWebModule.ts
@@ -7,13 +7,18 @@ import type { NativeModule } from './ts-declarations/NativeModule';
  * @returns A singleton instance of the class passed into arguments.
  */
 
+type ModuleIdentifier = string | { name?: string };
+
 export function registerWebModule<
   EventsMap extends Record<never, never>,
   ModuleType extends typeof NativeModule<EventsMap>,
->(moduleImplementation: ModuleType): ModuleType {
+>(moduleImplementation: ModuleType, moduleIdentifier?: ModuleIdentifier): ModuleType {
   ensureNativeModulesAreInstalled();
 
-  const moduleName = moduleImplementation.name;
+  const fallbackName =
+    typeof moduleIdentifier === 'string' ? moduleIdentifier : moduleIdentifier?.name;
+
+  const moduleName = moduleImplementation.name || fallbackName;
   if (!moduleName) {
     throw new Error('Module implementation must be a class');
   }
